### PR TITLE
add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4

### PR DESCRIPTION
**Pull request by glaringlee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37610 add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4**

New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

### CODE REVIEW

File: .circleci/docker/build.sh
Comment (seemethere) [inline .circleci/docker/build.sh:90]: We should probably just standardized on using the latest version of Python wherever possible.

Can you update this to Python 3.8?

Comment (glaringlee) [reply .circleci/docker/build.sh:90]: Sure, will do

File: .circleci/verbatim-sources/workflows-docker-builder.yml
Comment (ailzhang) [inline .circleci/verbatim-sources/workflows-docker-builder.yml:2]: These changes need to be restored before merging.

File: .circleci/verbatim-sources/workflows-ecr-gc.yml
Comment (ailzhang) [inline .circleci/verbatim-sources/workflows-ecr-gc.yml:2]: And this one

File: .circleci/verbatim-sources/workflows-docker-builder.yml
Comment (glaringlee) [reply .circleci/verbatim-sources/workflows-docker-builder.yml:2]: Yep, I was trying to switch to a python 3.8 image, will add them back later

File: .circleci/docker/build.sh
Comment (glaringlee) [reply .circleci/docker/build.sh]: @seemethere 
I tried to build python 3.8 image, it seems to me that there is no simple way at this point to create such docker image. 
There are 2 things that blocking it:

1. Pytorch docker image install llvmlite 0.30.0 which does not support llvm9 under python 3.8, it need llvm-config which is not available in llvm9. 

2. Yes, I tried switching to llvmlite 0.31.0 (or 32), it works, the docker image can be built, but the gcc-5 version is upgraded to 5.5 instead of 5.4, this will affect all the pytorch docker image that using gcc-5, and currently, all the gcc-5 pytorch CI test assume it uses gcc 5.4,  so I think it would be better to make such change in a separate issue since this is not trivial.

I'd like to keep using python3.6 for this image, please advise.

Comment (seemethere) [reply .circleci/docker/build.sh]: Sure let's stick to Python 3.6 here. Can you create a tracking issue to update these images to Python 3.8 with this information inside of the issue?

**Review by seemethere — COMMENTED** [review]
LGTM after we restore the triggers

File: .circleci/docker/build.sh
Comment (glaringlee) [reply .circleci/docker/build.sh]: sure. Created https://github.com/pytorch/pytorch/issues/37847
Will go back to 3.6 and request review agan.

**Comment by glaringlee** [comment]
@seemethere @ailzhang 
restored to 3.6 and rebased master. please check and stamp if looks good. thx.

**Review by seemethere — APPROVED** [review]

**Review by ailzhang — APPROVED** [review]
LGTM!

**Pull request closed without merging**
